catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,265 @@
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.source.Source;
+
+public class RegistryPublicationActionProvider implements ActionProvider, EventHandler {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(RegistryPublicationActionProvider.class);
+
+    private static final String REGISTRY_PATH = "/registries";
+
+    private static final String PUBLICATION_PATH = "publication";
+
+    private static final String PUBLISH_DESCRIPTION = "Publishes this nodes information to ";
+
+    private static final String PUBLISH_OPERATION = "publish";
+
+    private static final String PUBLISH_TITLE = "Publish to ";
+
+    private static final String UNPUBLISH_DESCRIPTION = "Unpublishes this nodes information from ";
+
+    private static final String UNPUBLISH_OPERATION = "unpublish";
+
+    private static final String UNPUBLISH_TITLE = "Unpublish from ";
+
+    private static final String HTTP_POST = "HTTP_POST";
+
+    private static final String HTTP_DELETE = "HTTP_DELETE";
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private static final String CREATED_TOPIC = "ddf/catalog/event/CREATED";
+
+    private static final String UPDATED_TOPIC = "ddf/catalog/event/UPDATED";
+
+    private static final String DELETED_TOPIC = "ddf/catalog/event/DELETED";
+
+    private String providerId;
+
+    private ConfigurationAdmin configAdmin;
+
+    private FederationAdminService federationAdminService;
+
+    private List<RegistryStore> registryStores;
+
+    private Map<String, List<String>> publications = new ConcurrentHashMap<>();
+
+    @Override
+    public <T> List<Action> getActions(T subject) {
+        String registryId = getRegistryId(subject);
+        if (registryId == null) {
+            return Collections.emptyList();
+        }
+        List<Action> actions = new ArrayList<>();
+        List<String> currentPublications = publications.get(registryId);
+
+        for (RegistryStore registry : registryStores) {
+            //can't publish to yourself
+            if (!registry.isPushAllowed() || StringUtils.isBlank(registry.getRegistryId())
+                    || registryId.equals(registry.getRegistryId())) {
+                continue;
+            }
+
+            if (currentPublications.contains(registry.getId())) {
+                CollectionUtils.addIgnoreNull(actions,
+                        getAction(registryId, registry.getId(), false));
+            } else {
+                CollectionUtils.addIgnoreNull(actions,
+                        getAction(registryId, registry.getId(), true));
+            }
```

Comment:
this could be done in one line currentPublications.contains(registry.getId()) can be used for the true false